torchgeo/datasets/cdl.py

```diff
@@ -363,15 +372,33 @@ def _verify(self) -> None:
             RuntimeError: if ``download=False`` but dataset is missing or checksum fails
         """
         # Check if the extracted files already exist
-        pathname = os.path.join(self.root, "**", self.filename_glob)
-        for fname in glob.iglob(pathname, recursive=True):
-            if not fname.endswith(".zip"):
-                return
+        exists = []
+        for year in self.years:
+            filename_year = self.filename_glob.replace("*", str(year))
+            pathname = os.path.join(
+                self.root, filename_year.removesuffix(".tif"), filename_year
```

Comment:
We should keep the recursive glob so that people can specify a higher or lower root directory (e.g., `root=/path/to/2019_30m_cdls/` or `root=/path/`)